src/main/java/org/terasology/gookeeper/system/EconomySystem.java

```diff
@@ -229,18 +230,18 @@ public void payVisitFee(VisitorComponent visitorComponent, EntityRef visitBlock)
             VisitorEntranceComponent visitorEntranceComponent =
                     visitorComponent.visitorEntranceBlock.getComponent(VisitorEntranceComponent.class);
             EntityRef player = visitorEntranceComponent.owner;
-            EconomyComponent economyComponent = player.getComponent(EconomyComponent.class);
 
             VisitBlockComponent visitBlockComponent = visitBlock.getComponent(VisitBlockComponent.class);
 
             Prefab gooeyPrefab = prefabManager.getPrefab("GooKeeper:" + visitBlockComponent.type);
 
-            if (economyComponent != null && gooeyPrefab != null && gooeyPrefab.hasComponent(GooeyComponent.class)) {
```

Comment:
Why did you remove the checks for `gooeyPrefab != null` and `gooeyPrefab.hasComponent(GooeyComponent.class)`?

This might still be breaking - is this such that it fails hard if these dependencies within the same module break?

Reply:
good catch... that wasn't on purpose tbh... and yes I think if the `getComponent` returns `null` and that is left unchecked, the resulting NPE would crash the game hard.